src/Prophecy.phpstub

```diff
@@ -0,0 +1,45 @@
+<?php
+namespace Prophecy\Prophecy {
+    /** @template T */
+    class ObjectProphecy implements ProphecyInterface
```

Comment:
This stub is missing `willExtendOrImplement` method. I think it should look like this:

```php
/**
 * @template U
 * @param class-string<U> $className
 * @return self<T&U>
 */
public function willExtendOrImplement(string $className)
{
}
```

I'm not sure it will work in current PHPStan, maybe it will need a fix. But the stub should definitely work like this.

Reply:
I don't see that method. I see two separate `willExtend` and `willImplement`, but both return `$this` (and are already correctly typed in the PHPDoc), not the mock.

Reply:
Yeah, there are two methods, sorry. But you actually need to add that type to the intersection :)

Reply:
> Yeah, there are two methods, sorry. But you actually need to add that type to the intersection :)

I did it here, is it enough?

https://github.com/Jan0707/phpstan-prophecy/pull/165/files#diff-79a44d595250a9b2a68a116ba32b375cR86